config/config.go

```diff
@@ -364,13 +367,20 @@ func ParseUpstream(upstream string) (Upstream, error) {
 	}
 
 	return Upstream{
-		Net:  n,
-		Host: host,
-		Port: port,
-		Path: path,
+		Net:        n,
+		Host:       host,
+		Port:       port,
+		Path:       path,
+		CommonName: commonName,
 	}, nil
 }
 
+func extractCN(in string) (cn string, upstream string) {
```

Comment:
Maybe we can rename the function to `extractCommonName` just to be extra clear for future readers.